File: algo/src/test/java/com/hartwig/actin/algo/evaluation/EvaluationAssert.java
package com.hartwig.actin.algo.evaluation;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;

import org.jetbrains.annotations.NotNull;

public final class EvaluationAssert {

    private EvaluationAssert() {
    }

    public static void assertEvaluation(@NotNull EvaluationResult expected, @NotNull Evaluation actual) {
        assertEquals(expected, actual.result());

        if (actual.result() == EvaluationResult.PASS) {
            assertFalse(actual.passSpecificMessages().isEmpty());
            assertTrue(actual.warnSpecificMessages().isEmpty());
            assertTrue(actual.warnGeneralMessages().isEmpty());
            assertTrue(actual.undeterminedSpecificMessages().isEmpty());
            assertTrue(actual.undeterminedGeneralMessages().isEmpty());
            assertTrue(actual.failSpecificMessages().isEmpty());
            assertTrue(actual.failGeneralMessages().isEmpty());
        } else if (actual.result() == EvaluationResult.WARN) {
            assertTrue(actual.passSpecificMessages().isEmpty());
            assertTrue(actual.passGeneralMessages().isEmpty());
            assertFalse(actual.warnSpecificMessages().isEmpty());
            assertTrue(actual.undeterminedSpecificMessages().isEmpty());
            assertTrue(actual.undeterminedGeneralMessages().isEmpty());
            assertTrue(actual.failSpecificMessages().isEmpty());
            assertTrue(actual.failGeneralMessages().isEmpty());
        } else if (actual.result() == EvaluationResult.UNDETERMINED) {
            assertTrue(actual.passSpecificMessages().isEmpty());
            assertTrue(actual.passSpecificMessages().isEmpty());
            assertTrue(actual.passGeneralMessages().isEmpty());
            assertTrue(actual.warnSpecificMessages().isEmpty());
            assertTrue(actual.warnGeneralMessages().isEmpty());
            assertFalse(actual.undeterminedSpecificMessages().isEmpty());
            assertTrue(actual.failSpecificMessages().isEmpty());
            assertTrue(actual.failGeneralMessages().isEmpty());
        } else if (actual.result() == EvaluationResult.FAIL) {
            assertTrue(actual.passSpecificMessages().isEmpty());
            assertTrue(actual.passGeneralMessages().isEmpty());
            assertTrue(actual.warnSpecificMessages().isEmpty());
            assertTrue(actual.warnGeneralMessages().isEmpty());
            assertTrue(actual.undeterminedSpecificMessages().isEmpty());
            assertTrue(actual.undeterminedGeneralMessages().isEmpty());
            assertFalse(actual.failSpecificMessages().isEmpty());
        }
    }
}

File: algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/EvaluationAssert.kt
package com.hartwig.actin.algo.evaluation

import com.hartwig.actin.datamodel.algo.Evaluation
import com.hartwig.actin.datamodel.algo.EvaluationResult
import org.assertj.core.api.Assertions.assertThat

object EvaluationAssert {

    fun assertMolecularEvaluation(expected: EvaluationResult, actual: Evaluation) {
        assertEvaluation(expected, actual)
        if (actual.result == EvaluationResult.PASS || actual.result == EvaluationResult.WARN) {
            assertThat(actual.inclusionMolecularEvents).isNotEmpty()
        } else {
            assertThat(actual.inclusionMolecularEvents).isEmpty()
        }
    }

    fun assertEvaluation(expected: EvaluationResult, actual: Evaluation) {
        assertThat(actual.result).isEqualTo(expected)
        when (actual.result) {
            EvaluationResult.PASS -> {
                assertThat(actual.passMessagesStrings()).isNotEmpty()
                assertThat(actual.warnMessagesStrings()).isEmpty()
                assertThat(actual.undeterminedMessagesStrings()).isEmpty()
                assertThat(actual.failMessagesStrings()).isEmpty()
            }

            EvaluationResult.WARN -> {
                assertThat(actual.passMessagesStrings()).isEmpty()
                assertThat(actual.warnMessagesStrings()).isNotEmpty()
                assertThat(actual.undeterminedMessagesStrings()).isEmpty()
                assertThat(actual.failMessagesStrings()).isEmpty()
            }

            EvaluationResult.UNDETERMINED -> {
                assertThat(actual.passMessagesStrings()).isEmpty()
                assertThat(actual.warnMessagesStrings()).isEmpty()
                assertThat(actual.undeterminedMessagesStrings()).isNotEmpty()
                assertThat(actual.failMessagesStrings()).isEmpty()
            }

            EvaluationResult.FAIL -> {
                assertThat(actual.passMessagesStrings()).isEmpty()
                assertThat(actual.warnMessagesStrings()).isEmpty()
                assertThat(actual.undeterminedMessagesStrings()).isEmpty()
                assertThat(actual.failMessagesStrings()).isNotEmpty()
            }

            else -> {}
        }
    }
}